### PR TITLE
use wechatprogram subpackge default and speedup launch time

### PR DESCRIPTION
**Pull request by novain1102**
Re: # https://github.com/cocos/3d-tasks/issues/16071

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/gfx/webgl/webgl-commands.ts [hunk 1]
```diff
@@ -2714,10 +2715,30 @@ export function WebGLCmdFuncCopyTexImagesToTexture (
...
-            // console.debug('Copying image to texture 2D: ' + region.texExtent.width + ' x ' + region.texExtent.height);
-            gl.texSubImage2D(gl.TEXTURE_2D, region.texSubres.mipLevel,
-                region.texOffset.x, region.texOffset.y,
-                gpuTexture.glFormat, gpuTexture.glType, texImages[n++]);
+            if (WECHAT_MINI_PROGRAM) {
+                // TODO: Property 'type' does not exist on type 'HTMLImageElement'.
+                // maybe we need a higher level implementation called `pal/image`, we provide `type` interface here.
+                // issue: https://github.com/cocos/cocos-engine/issues/14646
+                const texImg = texImages[n++] as any;
+                if (texImg.type !== 'undefined' && texImg.type === 'canvas') {
```
Comment (minggo) [inline cocos/gfx/webgl/webgl-commands.ts]: It is better to hack in platform layer.

File: templates/wechatprogram/project.config.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     },
     "compileType": "miniprogram",
     "libVersion": "2.19.4",
-    "appid": "wx160703e6ec9ed84e",
+    "appid": "wx2c815291f1aa4d1c",
```
Comment (minggo) [inline templates/wechatprogram/project.config.json:46]: Will it change every time?

Comment (novain1102) [reply templates/wechatprogram/project.config.json:46]: Just for testing on my device more convenient.

File: cocos/gfx/webgl/webgl-commands.ts (hunk 1 above)
Comment (novain1102) [reply cocos/gfx/webgl/webgl-commands.ts]: We do not have this platform layer yet. Similar treatment is here: https://github.com/cocos/cocos-engine/pull/14638/commits/c268b8982cb9975e95b5f1ef855a681eff82b059

Comment (PPpro) [inline cocos/gfx/webgl/webgl-commands.ts]:
```suggestion
                if (typeof texImg.type !== 'undefined' && texImg.type === 'canvas') {
```

Comment (minggo) [reply cocos/gfx/webgl/webgl-commands.ts]: Can hack in upper layer, not in gfx?

Comment (novain1102) [reply cocos/gfx/webgl/webgl-commands.ts]: move to PAL

Comment (novain1102) [reply cocos/gfx/webgl/webgl-commands.ts]: done.

File: pal/minigame/wechat_mini_program.ts [hunk 3]
```diff
@@ -173,4 +173,23 @@ if (systemInfo.platform === 'windows' && versionCompare(systemInfo.SDKVersion, '
     }
 }
 
+// HACK: adapt gl.texSubImage2D: gl.texSubImage2D do not support 2d canvas in wechat miniprogram
+// @ts-expect-error getApp defined in global
```
Comment (minggo) [inline pal/minigame/wechat_mini_program.ts]: We forbid to use `@ts-expect-error` now.

Comment (novain1102) [reply pal/minigame/wechat_mini_program.ts]: fixed.

**Review by minggo — APPROVED** [review]

File: templates/wechatprogram/game.ejs [hunk 4]
```diff
@@ -57,6 +58,7 @@ Page({
                     return firstScreen.setProgress(0.6).then(() => Promise.resolve(module));
                 }).then((cc) => {
                     require('./engine-adapter');
+                    require('src/chunks/bundle.js'); //bundle.js should reuqire after engine-adapter if use named register mode
```
Comment (PPpro) [inline templates/wechatprogram/game.ejs]: this is a hack, and it seems we don't know why we should require after engine-adapter for now

Comment (PPpro) [reply templates/wechatprogram/game.ejs]: `src/chunks/bundle.js` maybe not exist after building project

**Review by PPpro — CHANGES_REQUESTED** [review]

File: templates/wechatprogram/game.ejs (hunk 4 above)
Comment (novain1102) [reply templates/wechatprogram/game.ejs]: fixed. move to load-module file.

**Review by PPpro — APPROVED** [review]

**Pull request merged**